vm.go

```diff
@@ -786,8 +786,9 @@ func (vm *vm) peek() Value {
 func (vm *vm) saveCtx(ctx *vmContext) {
 	ctx.prg, ctx.stash, ctx.privEnv, ctx.newTarget, ctx.result, ctx.pc, ctx.sb, ctx.args =
 		vm.prg, vm.stash, vm.privEnv, vm.newTarget, vm.result, vm.pc, vm.sb, vm.args
-	if vm.getFuncName() != "" {
-		ctx.funcName = vm.getFuncName()
+	funcName := vm.getFuncName()
```

Comment:
One small callout, `funcName` shadows the funcName function above in the file. We also are shadowing with `funcName` in a separate method but just wanted to make sure this won't be a problem.

Reply:
I don't see any `funcName` function being shadowed. Do you mind sharing a link to the function you're referring to?

Reply:
You can call it `vmFuncName` but I think this is fine, also great find!

Reply:
> I don't see any `funcName` function being shadowed. Do you mind sharing a link to the function you're referring to?

It's here https://github.com/mongodb-forks/goja/blob/master/value.go#L1142 in a different file

Reply:
👍 I agree with Gabri